core/src/test/java/com/linecorp/armeria/ReconfigureTlsTest.java

```diff
@@ -0,0 +1,99 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.client.BlockingWebClient;
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.internal.common.util.SelfSignedCertificate;
+import com.linecorp.armeria.server.Server;
+
+class ReconfigureTlsTest {
+
+    @Test
+    void shouldUpdateTlsSettings() throws CertificateException {
+        final SelfSignedCertificate oldCert =
+                new SelfSignedCertificate(Date.from(Instant.parse("2022-01-01T00:00:00.00Z")),
+                                          Date.from(Instant.now().plus(10, ChronoUnit.DAYS)));
+        final AtomicReference<X509Certificate> sslContextRef = new AtomicReference<>();
+        final Server server =
+                Server.builder()
+                      .https(0)
+                      .tls(oldCert.certificate(), oldCert.privateKey())
+                      .service("/", (ctx, req) -> {
+                          sslContextRef.set((X509Certificate) ctx.sslSession().getLocalCertificates()[0]);
+                          return HttpResponse.of("OK");
+                      })
+                      .build();
+
+        server.start().join();
+
+        final BlockingWebClient client =
+                WebClient.builder("https://127.0.0.1:" + server.activeLocalPort())
+                         .factory(ClientFactory.builder()
+                                               .tlsCustomizer(sslContextBuilder -> {
+                                                   sslContextBuilder.trustManager(
+                                                           oldCert.certificate());
+                                               })
+                                               .build())
+                         .build()
+                         .blocking();
+        final AggregatedHttpResponse response = client.get("/");
+        assertThat(response.status()).isEqualTo(HttpStatus.OK);
+        assertThat(sslContextRef.get().getNotBefore()).isEqualTo(oldCert.cert().getNotBefore());
+
+        final SelfSignedCertificate newCert =
+                new SelfSignedCertificate(Date.from(Instant.parse("2023-01-01T00:00:00.00Z")),
```

Comment:
This will suddenly fail the test in 2024? Maybe use a relative time such as `now() +/- 10days`?

Reply:
I changed it to use a relative time. 👍